packages/appkit/src/client.ts

```diff
@@ -1249,21 +1249,25 @@ export class AppKit {
       )
     })
 
-    const namespace = ChainController.state.activeChain as ChainNamespace
+    const namespace = StorageUtil.getActiveNamespace()
+
     if (isConnected && this.connectionControllerClient?.connectExternal) {
       await this.connectionControllerClient?.connectExternal({
         id: ConstantsUtil.CONNECTOR_ID.AUTH,
         info: { name: ConstantsUtil.CONNECTOR_ID.AUTH },
         type: UtilConstantsUtil.CONNECTOR_TYPE_AUTH as ConnectorType,
         provider,
-        chainId: ChainController.state.activeCaipNetwork?.id
+        chainId: ChainController.state.activeCaipNetwork?.id,
+        chain: namespace
       })
       this.setLoading(false)
       this.setStatus('connected', namespace)
     } else {
       this.setLoading(false)
-      this.setStatus('disconnected', namespace)
-      StorageUtil.removeConnectedNamespace(namespace)
+      if (namespace) {
```

Comment:
Should we instead wrap parent block with this if, bc in all conditions (if and elses) we are using namespace, if it's not truthy, we shouldn't run any of those logics?

Reply:
Did and also abstracted all the event listeners to their own function, this was getting weird